### ISSUE TITLE
optimizer = auto problem.

### ISSUE THREAD

**Issue by sarpx** [ultralytics/ultralytics#21171]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

When we give optimizer = auto, we cannot see which optimizer it gives. Do you save this somewhere?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @sarpx, thank you for reaching out and using Ultralytics 🚀! We encourage you to check out the [Docs](https://docs.ultralytics.com/) for helpful [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) examples, as many common questions are already addressed there.

This is an automated response—an Ultralytics engineer will review your question and provide further assistance soon.

If this is a 🐛 Bug Report, please include a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us investigate.

For custom training or usage ❓ questions, please share as much detail as possible, including configuration, dataset examples, and relevant logs. Also, review our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Feel free to connect with the Ultralytics community:
- [Discord](https://discord.com/invite/ultralytics) for real-time support 🎧
- [Discourse](https://community.ultralytics.com/) for in-depth discussions
- [Subreddit](https://www.reddit.com/r/Ultralytics/) to share and learn

## Upgrade

Make sure you are using the latest `ultralytics` package with all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment and [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

YOLO can be run in these verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by NOTGOOOOD** [comment]
If set optimizer = 'auto', the optimizer is selected  based on the number of iterations. 

I remembered that if epoch>100, SGD will be used, otherwise AdamW will be used. You should verify this issue in the source code.

By the way, which optimizer you are using must print in the console log.

**Comment by sarpx** [comment]
@NOTGOOOOD Yes, what you say is very true, but I am trying to analyze the values ​​in different trainings, I need clearer information. If so, take this, if so, take this, I don't want to do it, and I also expect the optimizer it uses as logic to be written somewhere because this information is very important.

**Comment by Y-T-G** [comment]
It should show the optimizer in logs before training starts

**Comment by sarpx** [comment]
@Y-T-G  I know that the optimizer came out when the training started. But the problem is that months pass and two different models need to be evaluated and a comparison needs to be made to understand why it exploded or why it gave good results, and since we cannot access the optimizer directly, we cannot be sure what it did.
